libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -0,0 +1,158 @@
+ * - `switch` (default): will cancel the previous computation when a new one is triggered
+ * - `merge`: will use `mergeMap` to merge the last observable with the new one
+ * - `concat`: will use `concatMap` to concat the last observable with the new one
+ * - `exhaust`: will use `exhaustMap` to skip all the new emissions until the last observable completes
+ *
+ * You can also pass an `injector` option if you want to use it outside of the injection context.
+ *
+ * @param computation
+ * @param options
+ */
+export function computedAsync<T>(
+	computation: (previousValue?: T | undefined) => ComputationResult<T>,
+	options?: ComputedAsyncOptions<T>,
+): Signal<T> {
+	return assertInjector(computedAsync, options?.injector, () => {
+		const destroyRef = inject(DestroyRef);
+
+		// source$ is a Subject that will emit the new source value
+		const sourceEvent$ = new Subject<Promise<T> | Observable<T>>();
+
+		// will hold the current value
+		const sourceValue = signal<T | undefined>(
+			options?.initialValue ?? undefined,
+		);
+
+		const effectRef = effect(
+			() => {
+				// we need to have an untracked() here because we don't want to register the sourceValue as a dependency
+				// otherwise, we would have an infinite loop
+				const currentValue = untracked(() => sourceValue());
+
+				const newSource = computation(currentValue);
+				if (!isObservable(newSource) && !isPromise(newSource)) {
+					// if the new source is not an observable or a promise, we set the value immediately
+					untracked(() => sourceValue.set(newSource));
+					return;
+				}
+
+				// we untrack the source$.next() so that we don't register other signals as dependencies
+				untracked(() => sourceEvent$.next(newSource));
+			},
+			{ injector: options?.injector },
+		);
+
+		const source$: Observable<T> = createFlattenObservable(
+			sourceEvent$,
+			options?.behavior ?? 'switch',
+		);
+
+		const sourceResult = source$.subscribe({
+			next: (value) => sourceValue.set(value),
+			error: (error) => {
+				// NOTE: Error should be handled by the user (using catchError or .catch())
+				sourceValue.set(error);
+				throw error;
+			},
+		});
+
+		destroyRef.onDestroy(() => {
+			effectRef.destroy();
+			sourceResult.unsubscribe();
+		});
+
+		// we return a computed value that will return the current value
+		// in order to support the same API as computed()
+		return computed(
+			() => {
+				const value: T = sourceValue()!;
+				return value;
+			},
+			{ equal: options?.equal },
+		);
+	});
+}
+
+function createFlattenObservable<T>(
+	source: Subject<Promise<T> | Observable<T>>,
+	behavior: ComputedAsyncBehavior,
+): Observable<T> {
+	switch (behavior) {
+		case 'merge':
+			return source.pipe(mergeMap((s) => s));
+		case 'concat':
+			return source.pipe(concatMap((s) => s));
+		case 'exhaust':
+			return source.pipe(exhaustMap((s) => s));
+		default: // switch
+			return source.pipe(switchMap((s) => s));
+	}
+}
```

Comment:
I think `mergeAll`, `switchAll`, `exhaustAll` and `concatAll` can be used instead of `mergeMap((s) => s`, ...

Reply:
Hi, thanks.
But, there will be only one observable at a time? How will these other operators change anything?

Reply:
Hi! It doesn't mean that you flatten multiple observables but that it will flattne every observables going through that stream.

You wrote (let's take swtichMap as an example):
```ts
source.pipe(switchMap((s) => s));
```

But `(s) => s` can be "shortened" with [`identity`](https://rxjs.dev/api/index/function/identity)
```ts
source.pipe(switchMap(identity)); // which is the same as source.pipe(switchMap((s) => s))
```
And `swithMap(identity)` is the [`swithchAll`](https://github.com/ReactiveX/rxjs/blob/7f36e90626ba7f93687e227f8b2bf3c0302ac3ee/packages/rxjs/src/internal/operators/switchAll.ts#L62) operator
```ts
source.pipe(switchAll()); // which is the same as source.pipe(switchMap(identity))
```

Reply:
Nice! I learned something new and just updated the PR. Thanks 🤩

Reply:
Oh nice @LcsGa I didn't know it either. Thanks 🙏

Reply:
You're welcome 😁